package/libbatadv/src/batadv-genl.h

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: MIT */
+/* batman-adv helpers functions library
+ *
+ * Copyright (c) 2017, Sven Eckelmann <sven@narfation.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef _BATADV_GENL_H_
+#define _BATADV_GENL_H_
+
+#include <netlink/netlink.h>
+#include <netlink/genl/genl.h>
+#include <netlink/genl/ctrl.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "batman_adv.h"
+
+struct ether_addr;
+
+/**
+ * struct batadv_nlquery_opts - internal state for batadv_genl_query()
+ *
+ * This structure should be used as member of a struct which tracks the state
+ * for the callback. The macro container_of can be used to convert the
+ * arg pointer from batadv_nlquery_opts to the member which contains this
+ * struct.
+ */
+struct batadv_nlquery_opts {
+	/** @err: current error  */
+	int err;
+};
+
+/**
+ * ARRAY_SIZE() - Get number of items in static array
+ * @x: array with known length
+ *
+ * Return:  number of items in array
+ */
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof(*(x)))
+#endif
```

Comment:
I'm not happy with defining a macro with such a generic in a library header file. We should either give it a proper namespace prefix, or keep the duplication in the files using it.